scripts/import_cldr.py

```diff
@@ -286,15 +287,20 @@ def parse_global(srcdir, sup):
         region_code = region.attrib['iso3166']
         region_currencies = []
         for currency in region.findall('./currency'):
+            cur_code = currency.attrib['iso4217']
             cur_start = _parse_currency_date(currency.attrib.get('from'))
             cur_end = _parse_currency_date(currency.attrib.get('to'))
-            region_currencies.append((currency.attrib['iso4217'],
-                                      cur_start, cur_end,
-                                      currency.attrib.get(
-                                          'tender', 'true') == 'true'))
+            cur_tender = currency.attrib.get('tender', 'true') == 'true'
+            # Tie region to currency.
+            region_currencies.append((cur_code, cur_start, cur_end, cur_tender))
+            # Keep a reverse index of currencies to territorie.
+            all_currencies.setdefault(cur_code, []).append(region_code)
         region_currencies.sort(key=_currency_sort_key)
         territory_currencies[region_code] = region_currencies
 
+    # Deduplicate regions and makes them pickable.
+    all_currencies = dict([(code, tuple(set(regions))) for code, regions in all_currencies.items()])
```

Comment:
To be honest, it looks like that test is broken, then.
`__builtin__.set` should be available alright on both Python 2 and Python 3.
(I think what's happening is that there is no builtin operation in the pickle format for `set`s, so it resorts to "let's construct a class instance of `__builtin__.set`" instead.)

Anyway, fair! How about

```python
all_currencies = collections.defaultdict(set)
for currency in region.findall('./currency'):
    # ...
    all_currencies[cur_code].add(region_code)
    # ...
global_data['all_currencies'] = {
    currency: tuple(sorted(regions))
    for (currency, regions)
    in all_currencies.items()
}
```
?